### PR TITLE
Use new, separate `loopbreaker` library for inlining

### PR DESCRIPTION
**Pull request by TheMatten**
Red diffs, hooray! :tada:

TODO:
- [x] GHC 8.4 (guard use of plugin)

### CODE REVIEW

**Comment by isovector** [comment]
Hooray!!!

File: package.yaml [hunk 1]
```diff
@@ -69,7 +72,7 @@ library:
         - dump-core
       ghc-options:
         - -fplugin=DumpCore
-        - -fplugin-opt DumpCore:core-html
+        - -fplugin-opt=DumpCore:core-html
```
Comment (isovector) [inline package.yaml]: I am reasonably sure this works without the change

File: polysemy-plugin/src/Polysemy/Plugin.hs [hunk 2]
```diff
@@ -100,12 +94,9 @@ installTodos todos = do
     0 -> pure todos
     _ -> do
       mods <- moduleSetElts <$> getVisibleOrphanMods
-      pure $ case any ((== polysemyInternal) . moduleName) mods of
-        True  -> CoreDoPluginPass "Inline Recursive Calls" inlineRecursiveCalls
-               : todos
+      pure $ todos
```
Comment (isovector) [inline polysemy-plugin/src/Polysemy/Plugin.hs]: This change makes `mods` unused in the < `810` case, right? That'll make lint unhappy

File: src/Polysemy/Writer.hs [hunk 3]
```diff
@@ -53,7 +53,7 @@ runWriter = runState mempty . reinterpretH
         modify (`mappend` o) >>= pureT
       Listen m -> do
         mm <- runT m
-        -- TODO(sandy): this is fucking stupid
+        -- TODO(sandy): this is stupid
```
Comment (isovector) [inline src/Polysemy/Writer.hs]: :angry:

**Comment by isovector** [comment]
This brings true joy to my heart.

File: package.yaml (hunk 1 above)
Comment (TheMatten) [reply package.yaml]: I wanted to be consistent and `=` was just my personal preference - should I discard this change?

File: polysemy-plugin/src/Polysemy/Plugin.hs (hunk 2 above)
Comment (TheMatten) [reply polysemy-plugin/src/Polysemy/Plugin.hs]: Good point, thanks.

File: package.yaml (hunk 1 above)
Comment (isovector) [reply package.yaml]: I'm happy with it so long as it works.

File: polysemy-plugin/src/Polysemy/Plugin.hs [hunk 4]
```diff
@@ -93,10 +93,11 @@ installTodos todos = do
   case optLevel dflags of
     0 -> pure todos
     _ -> do
-      mods <- moduleSetElts <$> getVisibleOrphanMods
-      pure $ todos
 #if __GLASGOW_HASKELL__ >= 810
-          ++ bool []
-                  (extraPhases dflags)
-                  (any ((== polysemyInternal) . moduleName) mods)
+      mods <- moduleSetElts <$> getVisibleOrphanMods
```
Comment (isovector) [inline polysemy-plugin/src/Polysemy/Plugin.hs]: Just to ease my mind, did you check this branch to make sure it compiles?

Comment (TheMatten) [reply polysemy-plugin/src/Polysemy/Plugin.hs]: Yes - is there something that seems questionable?

Comment (isovector) [reply polysemy-plugin/src/Polysemy/Plugin.hs]: Nope, nothing! Just being overly cautious!

**Comment by isovector** [comment]
I'm happy to merge when you give me the :+1:

File: package.yaml [hunk 5]
```diff
@@ -43,10 +43,14 @@ default-extensions:
   - UnicodeSyntax
 
 when:
-  - condition: impl(ghc < 8.6)
+  condition: impl(ghc < 8.6)
+  then:
     default-extensions:
       - MonadFailDesugaring
       - TypeInType
+  else:
```
Comment (isovector) [inline package.yaml]: Let's write this as a separate condition, since it's not actually related to the `else` branch.

**Comment by TheMatten** [comment]
CI passed - :+1:

**Pull request merged**